configure.pl

```diff
@@ -514,165 +520,33 @@
 
-#
-# clean pid and sock files from old location (they are now in $Conf{RunDir}, and they
-# get re-created each time BackupPC starts, so it's ok if RunDir eq LogDir).
-#
-unlink("$DestDir$Conf{LogDir}/BackupPC.pid")  if ( -f "$DestDir$Conf{LogDir}/BackupPC.pid" );
-unlink("$DestDir$Conf{LogDir}/BackupPC.sock") if ( -e "$DestDir$Conf{LogDir}/BackupPC.sock" );
-
-if ( $Conf{CgiImageDir} ne "" ) {
-    printf("Installing images in $DestDir$Conf{CgiImageDir}\n");
-    foreach my $img ( <images/*> ) {
-	(my $destImg = $img) =~ s{^images/}{};
-	InstallFile($img, "$DestDir$Conf{CgiImageDir}/$destImg", 0444, 1);
-    }
-
-    #
-    # Install new CSS file, making a backup copy if necessary
-    #
-    my $cssBackup = "$DestDir$Conf{CgiImageDir}/BackupPC_stnd.css.pre-__VERSION__";
-    if ( -f "$DestDir$Conf{CgiImageDir}/BackupPC_stnd.css" && !-f $cssBackup ) {
-	rename("$DestDir$Conf{CgiImageDir}/BackupPC_stnd.css", $cssBackup);
-    }
-    InstallFile("conf/BackupPC_stnd.css",
-	        "$DestDir$Conf{CgiImageDir}/BackupPC_stnd.css", 0444, 0);
-    InstallFile("conf/BackupPC_retro_v2.css",
-	        "$DestDir$Conf{CgiImageDir}/BackupPC_retro_v2.css", 0444, 0);
-    InstallFile("conf/BackupPC_retro_v3.css",
-	        "$DestDir$Conf{CgiImageDir}/BackupPC_retro_v3.css", 0444, 0);
-    InstallFile("conf/sorttable.js",
-                "$DestDir$Conf{CgiImageDir}/sorttable.js", 0444, 0);
-}
+EOF
 
-printf("Making systemd and init.d scripts\n");
-mkpath("systemd/init.d", 0, 0755);
-foreach my $init ( qw(backuppc.service init.d/gentoo-backuppc init.d/gentoo-backuppc.conf
-                      init.d/linux-backuppc init.d/solaris-backuppc init.d/debian-backuppc
-                      init.d/freebsd-backuppc init.d/freebsd-backuppc2 init.d/suse-backuppc
-                      init.d/slackware-backuppc init.d/ubuntu-backuppc ) ) {
-    InstallFile("systemd/src/$init", "systemd/$init", 0755);
-}
+exit unless prompt("--> Do you want to continue?", "y") =~ /y/i;
 
-printf("Making Apache configuration file for suid-perl\n");
-InstallFile("httpd/src/BackupPC.conf", "httpd/BackupPC.conf", 0644);
+CleanPidSock();
 
-printf("Installing docs in $DestDir$Conf{InstallDir}/share/doc/BackupPC\n");
-foreach my $doc ( qw(BackupPC.pod BackupPC.html) ) {
-    InstallFile("doc/$doc", "$DestDir$Conf{InstallDir}/share/doc/BackupPC/$doc", 0444);
-    #
-    # clean up files from old directory
-    #
-    unlink("$DestDir$Conf{InstallDir}/doc/$doc") if ( -f "$DestDir$Conf{InstallDir}/doc/$doc" );
-}
-#
-# clean up old directory (ok if it quietly fails if there are other files in that directory)
-#
-rmdir("$DestDir$Conf{InstallDir}/doc") if ( -d "$DestDir$Conf{InstallDir}/doc" );
+DoInstall()
+  unless $opts{"config-only"};
 
 printf("Installing config.pl and hosts in $DestDir$Conf{ConfDir}\n");
-InstallFile("conf/hosts", "$DestDir$Conf{ConfDir}/hosts", 0644)
+InstallFile("conf/hosts", "$DestDir$Conf{ConfDir}/hosts.sample", 0644)
```

Comment:
This is all about making `configure.pl` more package friendly.

1. In general the `configure.pl` script is executed at package creation time and installs the program into a staging directory.
2. The preinstalled program files distributed as a package. A package installation is a basically copying that files into proper places on the hosts file sysytem.

3. After the package is installed user needs to run `configure.pl` again to upgrade `config.pl` from previous version (and maybe to configure new installation).
This time we should use `--config-only` option to skip installation phase.

Since `configure.pl` uses `conf/config.pl` and BackupPC source code or distribution is not available, we should add `conf/config.pl` to the package and install it somewhere.

It is a common good practice to provide sample configuration files that could be used by user as a reference or to fall back to default configuration.
Basically they are copies of configuration files installed by a package.

My idea is we can have `config.pl.sample` for both [3] and as a sample. That is why it gets created if !$opts{"config-only"} and used if !$opts{"config-only"}.

`hosts.sample` doesn't needed for `configure.pl`, it is just a sample file. So, it should be created only once at installation time.

I know it is not an elegant solution, but it doesn't involves major rewriting of the `configure.pl` and will substantially simplify creation of BackupPC packages. That is very complicated for now since `configure.pl` makes modifications at once in program and configuration files.